homeassistant/components/feedreader/coordinator.py

```diff
@@ -0,0 +1,247 @@
+        elif "published_parsed" in entry and entry.published_parsed:
+            # We are lucky, `published_parsed` data available, let's make use of
+            # it to publish only new available entries since the last run
+            self._has_published_parsed = True
+            self._last_entry_timestamp = max(
+                entry.published_parsed, self._last_entry_timestamp
+            )
+        else:
+            self._has_updated_parsed = False
+            self._has_published_parsed = False
+            _LOGGER.debug(
+                "No updated_parsed or published_parsed info available for entry %s",
+                entry,
+            )
+        entry.update({"feed_url": self._url})
+        self.hass.bus.fire(self._event_type, entry)
+        _LOGGER.debug("New event fired for entry %s", entry.get("link"))
+
+    def _publish_new_entries(self) -> None:
+        """Publish new entries to the event bus."""
+        assert self._feed is not None
+        new_entry_count = 0
+        self._last_entry_timestamp = self._storage.get_timestamp(self._feed_id)
+        if self._last_entry_timestamp:
+            self._firstrun = False
+        else:
+            # Set last entry timestamp as epoch time if not available
+            self._last_entry_timestamp = dt_util.utc_from_timestamp(0).timetuple()
+        # locally cache self._last_entry_timestamp so that entries published at identical times can be processed
+        last_entry_timestamp = self._last_entry_timestamp
+        for entry in self._feed.entries:
+            if (
+                self._firstrun
+                or (
+                    "published_parsed" in entry
+                    and entry.published_parsed > last_entry_timestamp
+                )
+                or (
+                    "updated_parsed" in entry
+                    and entry.updated_parsed > last_entry_timestamp
+                )
+            ):
+                self._update_and_fire_entry(entry)
+                new_entry_count += 1
+            else:
+                _LOGGER.debug("Already processed entry %s", entry.get("link"))
+        if new_entry_count == 0:
+            self._log_no_entries()
+        else:
+            _LOGGER.debug("%d entries published in feed %s", new_entry_count, self._url)
+        self._firstrun = False
+
+
+class StoredData:
+    """Represent a data storage."""
+
+    def __init__(self, hass: HomeAssistant, legacy_data_file: str) -> None:
+        """Initialize data storage."""
+        self._legacy_data_file = legacy_data_file
+        self._data: dict[str, struct_time] = {}
+        self.hass = hass
+        self._store: Store[dict[str, str]] = Store(hass, STORAGE_VERSION, DOMAIN)
+
+    async def async_setup(self) -> None:
+        """Set up storage."""
+        if not os.path.exists(self._store.path):
+            # Remove the legacy store loading after deprecation period.
+            data = await self.hass.async_add_executor_job(self._legacy_fetch_data)
```

Comment:
How long is the legacy store deprecated? Maybe it's time to remove it

Reply:
it was migrated in #98754 (_released with 2023.9_) and "announced" with a 6 months deprecation period - so we could remove it now. I'll try to remove it

Reply:
3851c2cefbddf514918d459a3c61bde6d57c29a3